pkg/cache/placeholder_manager.go

```diff
@@ -61,6 +61,11 @@ func NewPlaceholderManager(clients *client.Clients) *PlaceholderManager {
 }
 
 func getPlaceholderManager() *PlaceholderManager {
+	placeholderMgr.RLock()
+	defer placeholderMgr.RUnlock()
+	if placeholderMgr == nil {
+		return nil
```

Comment:
need to put the nil check upfront

```
if placeholderMgr == nil {
		return nil
}
placeholderMgr.RLock()
defer placeholderMgr.RUnlock()
...

```

Reply:
but it will cause other data race, when we check the nil, need to read the placeholderMgr.